core/classes/CaptchaBase.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/*
+ *	Made by Samerton
+ *
+ *  https://github.com/NamelessMC/Nameless/
+ *  NamelessMC version 2.0.0-pr10
+ *
+ *  License: MIT
+ *
+ *  CaptchaBase class
+ */
+abstract class CaptchaBase {
+    protected static $_providers = array(), $_activeProvider = '';
+
+    protected $_name, $_publicKey, $_privateKey;
+
+    /**
+     * Register a provider
+     * @param CaptchaBase $provider Provider instance to register
+     */
+    public static function addProvider(CaptchaBase $provider) {
+        CaptchaBase::$_providers[$provider->_name] = $provider;
```

Comment:
Should we use `self` here for consistency with the rest of the class?

Reply:
Yep good catch, too used to TS... 😄 

In terms of the blank line, personally I'm not a fan of blank lines under class/method declarations - I just feel like they're unnecessary since the declaration is already pretty short and unobtrusive.